data/physics_based_pi_formatter.py

```
@@ -0,0 +1,800 @@
+                expected_flux = self.neutronics_model.calculate_neutron_flux_from_power(
+                    power_level if power_level <= 100 else power_level/30.0  # Handle MW vs % units
+                )
+                flux_deviation = abs(neutron_flux - expected_flux) / expected_flux
+                
+                if flux_deviation > self.config.power_flux_tolerance:
+                    issues.append(f"Power-flux correlation off by {flux_deviation*100:.1f}%")
+                    severity = 'questionable' if flux_deviation > 0.3 else 'uncertain'
+            
+            if rod_position is not None and power_level is not None:
+                # Validate rod position vs power correlation
+                expected_reactivity = self.neutronics_model.calculate_control_rod_reactivity(rod_position)
```

Ruff (0.11.9)

Local variable `expected_reactivity` is assigned to but never used

Remove assignment to unused variable `expected_reactivity`

(F841)